src/array.ts

```diff
@@ -1,31 +1,27 @@
-export const objectListForAttribute = <T>(
+export const groupByKey = <T>(
   objects: T[],
-  attribute: string
+  key: string
```

Comment:
It's usually called an [object key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) so I thought that would be more descriptive.